langchain4j-qianfan/src/main/java/dev/langchain4j/model/qianfan/InternalQianfanHelper.java

```diff
@@ -0,0 +1,221 @@
+package dev.langchain4j.model.qianfan;
+
+import dev.langchain4j.agent.tool.ToolExecutionRequest;
+import dev.langchain4j.agent.tool.ToolParameters;
+import dev.langchain4j.agent.tool.ToolSpecification;
+import dev.langchain4j.data.message.*;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.internal.Utils;
+import dev.langchain4j.model.qianfan.client.embedding.EmbeddingResponse;
+import dev.langchain4j.model.qianfan.client.chat.Parameters;
+import dev.langchain4j.model.output.FinishReason;
+import dev.langchain4j.model.output.TokenUsage;
+import dev.langchain4j.model.qianfan.client.chat.ChatCompletionResponse;
+import dev.langchain4j.model.qianfan.client.chat.Message;
+import dev.langchain4j.model.qianfan.client.chat.Role;
+import dev.langchain4j.model.qianfan.client.chat.FunctionCall;
+import dev.langchain4j.model.qianfan.client.chat.Function;
+import dev.langchain4j.model.qianfan.client.completion.CompletionResponse;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import static dev.langchain4j.data.message.AiMessage.aiMessage;
+import static dev.langchain4j.internal.Exceptions.illegalArgument;
+import static dev.langchain4j.model.output.FinishReason.*;
+import static java.util.stream.Collectors.toList;
+
+public class InternalQianfanHelper {
+
+    public static List<Function> toFunctions(Collection<ToolSpecification> toolSpecifications) {
+        return toolSpecifications.stream()
+                .map(InternalQianfanHelper::toFunction)
+                .collect(toList());
+    }
+
+    private static Function toFunction(ToolSpecification toolSpecification) {
+        return Function.builder()
+                .name(toolSpecification.name())
+                .description(toolSpecification.description())
+                .parameters(toOpenAiParameters(toolSpecification.parameters()))
+                .build();
+    }
+
+    private static Parameters toOpenAiParameters(ToolParameters toolParameters) {
+        if (toolParameters == null) {
+            return Parameters.builder().build();
+        }
+        return Parameters.builder()
+                .properties(toolParameters.properties())
+                .required(toolParameters.required())
+                .build();
+    }
+    public static Message toErnieBotMessage(ChatMessage message) {
```

Comment:
Please use consistent naming. Right now there are already 4: "qianfan", "ernie", "baidu", "wenxin"...

Reply:
I will change the variable name to qianfan